packages/cli/src/cmds/validator/voluntaryExit.ts

```diff
@@ -106,53 +114,67 @@ ${validatorsToExit.map((v) => `${v.pubkey} ${v.index} ${v.status}`).join("\n")}`
     }
 
     for (const [i, {index, signer, pubkey}] of validatorsToExit.entries()) {
-      const domain = config.getDomainForVoluntaryExit(computeStartSlotAtEpoch(exitEpoch));
+      const slot = computeStartSlotAtEpoch(exitEpoch);
+      const domain = config.getDomainForVoluntaryExit(slot);
       const voluntaryExit: phase0.VoluntaryExit = {epoch: exitEpoch, validatorIndex: index};
       const signingRoot = computeSigningRoot(ssz.phase0.VoluntaryExit, voluntaryExit, domain);
 
+      let signature;
+      switch (signer.type) {
+        case SignerType.Local:
+          signature = signer.secretKey.sign(signingRoot);
+          break;
+        case SignerType.Remote: {
+          const signatureHex = await externalSignerPostSignature(config, signer.url, pubkey, signingRoot, slot, {
+            data: voluntaryExit,
+            type: SignableMessageType.VOLUNTARY_EXIT,
+          });
+          signature = bls.Signature.fromHex(signatureHex);
+          break;
+        }
+        default:
+          throw new YargsError(`Unexpected signer type for ${pubkey}`);
+      }
       ApiError.assert(
         await client.beacon.submitPoolVoluntaryExit({
           message: voluntaryExit,
-          signature: signer.secretKey.sign(signingRoot).toBytes(),
+          signature: signature.toBytes(),
         })
       );
-
       console.log(`Submitted voluntary exit for ${pubkey} ${i + 1}/${signersToExit.length}`);
     }
   },
 };
 
-type SignerLocalPubkey = {signer: SignerLocal; pubkey: string};
+type SignerPubkey = {signer: SignerLocal | SignerRemote; pubkey: string};
```

Comment:
Can reuse existing type here

https://github.com/ChainSafe/lodestar/blob/8afbc98b561d714fdadd1946ec2a5aab83f479de/packages/validator/src/services/validatorStore.ts#L115

```suggestion
type SignerPubkey = {signer: Signer; pubkey: string};
```